### DISCUSSION TITLE
Generic IndexedDB Observer

### DISCUSSION THREAD

**Discussion by KooIaIa** [appy-one/acebase#101] [Ideas]
I have a open source app that uses IndexedDB that's already written. I was wondering about using AceBase to sync its data to a server to make it multiplayer (its a game engine). Right now all data is stored in IndexedDB anytime something is saved.

![image](https://user-images.githubusercontent.com/30942833/167759792-ae3ea29d-e45d-44df-92d0-92bf59b5d9ae.png)

I was wondering about making a AceBase database line up to overlap with the existing IndexedDB database so everything in it could be accessed from Acebase.

When I create a new database with AceBase.WithIndexedDB('mydatabase') it appends .acebase to the IndexedDB name so I would need to edit that function (maybe with a optional setting) so that their names line up and the data overlaps.

Is this idea impossible though, because for Acebase to function it needs an additional **_nodes_** IndexedDB entry and the only way that happens is if data is called through the Acebase API?

Is there maybe some clever way to use Acebase with applications putting data into IndexedDB directly without re-writing the app to use Acebase?

**Comment by appy-one** [comment]
You can't. AceBase supports storing tree-like data structures and allows selectively loading nodes from anywhere in that tree. Because IndexedDB is a key-value store, AceBase has to jump through a number of hoops to flatten that tree into key-value pairs with contextual info about their whereabouts in the tree. Also, IndexedDB does not raise events if data is changed, so you can't monitor a database's content and anticipate on those changes. You'd have to change your code to use AceBase.
